interop-cats/jvm/src/main/scala/scalaz/zio/interop/catszmanagedjvm.scala

```diff
@@ -0,0 +1,102 @@
+package scalaz.zio.interop
+
+import cats.effect.{ Bracket, LiftIO, Resource }
+import cats.{ ~>, effect, Bifunctor, Defer, Functor, Monad, MonadError, Monoid, Semigroup, SemigroupK }
+import scalaz.zio.{ Reservation, ZIO, ZManaged }
+
+trait CatsZManagedSyntax {
+
+  implicit class ResourceSyntax[F[_], A](resource: Resource[F, A]) {
+    def toManaged(
+      implicit nat: F ~> ZIO[Any, Throwable, ?],
+      b: Bracket[F, Throwable],
+      zB: Bracket[ZIO[Any, Throwable, ?], Throwable],
+      d: Defer[ZIO[Any, Throwable, ?]]
+    ): ZManaged[Any, Throwable, A] =
+      ZManaged.unwrap(
+        resource
+          .mapK(nat)
+          .allocated
+          .map {
+            case (acquire, cleanup) =>
+              val reservation = Reservation(ZIO.succeed(acquire), cleanup.orDie) // should we die here or swallow the exception with catchall, etc?
+              ZManaged.reserve(reservation)
+          }
+          .uninterruptible
+      )
+  }
+
+  implicit class ZManagedSyntax[R, A](managed: ZManaged[Any, Throwable, A]) {
+    def toResource[F[_]: Functor](implicit nat: ZIO[Any, Throwable, ?] ~> F): Resource[F, A] = {
```

Comment:
Instead of requiring the implicit natural transformation, try requiring `implicit R: Runtime[Any]`. In fact, then you can generalize to support `ZManaged[R, ...`, `ZIO[R, ...`.

Reply:
You are definitely right that this can be generalized. I updated the pr.

I would still need something to go from F to ZIO, no? I think everything I could do here is just special case something that is better handled as an instance of a natural transformation in the zio cats interop. I can see whether I can define something from ZIO[Any, x, x] without requiring an implicit runtime and one from ZIO[R, x, x] using an implicit runtime.

Not fixing this to throwable has the disadvantage of not being able to call orDie on the release. Which would only leave swallowing the exception. I don't know whether this is a good tradeoff.

I can rewrite the logic to not require the bracket instance Bracket[ZIO[R, E, A], Throwable] but this then instead requires a Functor[F]. Also actually calling use on the Resource would still require Bracket[ZIO[R, E, A], Throwable], for which I can't imagine a sane implementation where E isn't throwable.

Reply:
The implicit natural transformation pattern was used in Cats `STM`, but since natural transformations are not unique nor type classes, it doesn't make sense to use them in this fashion; also it was possible to replace it by an implicit `Runtime` in Cats `STM` (you can look at the techniques there), which is more sound since that class is designed to be implicit.

Reply:
Ok. I'll try to copy what is done there.
Thanks for your comments